scripts/build-windows-k8s.sh

```diff
@@ -370,12 +346,74 @@ cleanup_output() {
 	rm -r ${TOP_DIR}
 }
 
-create_dist_dir
-get_kube_binaries
-download_nssm
-download_wincni
-copy_dockerfile
-create_zip
-upload_zip_to_blob_storage
-push_acs_branch
-cleanup_output
+
+ACS_ENGINE_HOME=${GOPATH}/src/github.com/Azure/acs-engine
+
+usage() {
+	echo "$0 [-v version] [-p acs_patch_version]"
+	echo " -v <version>: version"
+	echo " -p <patched version>: acs_patch_version"
+	echo " -u <version build for kubernetes upstream e2e tests>: k8s_e2e_upstream_version"
+	echo " -z <zip path>: zip_path"
+}
+
+while getopts ":v:p:u:z:" opt; do
+  case ${opt} in
+    v)
+      version=${OPTARG}
+      ;;
+    p)
+      acs_patch_version=${OPTARG}
+      ;;
+	u)
+	  k8s_e2e_upstream_version=${OPTARG}
+	  ;;
+	z)
+	  zip_path=${OPTARG}
+	  ;;  
+    *)
+			usage
+			exit
+      ;;
+  esac
+done
+
+KUBEPATH=${GOPATH}/src/k8s.io/kubernetes
+
+if [ -z "${k8s_e2e_upstream_version}" ]; then
+
+	if [ -z "${version}" ] || [ -z "${acs_patch_version}" ]; then
+		usage
+			exit 1
+	fi
+
+	if [ -z "${AZURE_STORAGE_CONNECTION_STRING}" ] || [ -z "${AZURE_STORAGE_CONTAINER_NAME}" ]; then
+		echo '$AZURE_STORAGE_CONNECTION_STRING and $AZURE_STORAGE_CONTAINER_NAME need to be set for upload to Azure Blob Storage.'
+			exit 1
+	fi
+
+	KUBERNETES_RELEASE=$(echo $version | cut -d'.' -f1,2)
+	KUBERNETES_TAG_BRANCH=v${version}
+	ACS_VERSION=${version}-${acs_patch_version}
+	ACS_BRANCH_NAME=acs-v${ACS_VERSION}
+	TOP_DIR=${ACS_ENGINE_HOME}/_dist/k8s-windows-v${ACS_VERSION}
+	DIST_DIR=${TOP_DIR}/k
+
+	create_dist_dir
+	get_kube_binaries
+	download_nssm
+	download_wincni
+	copy_dockerfile
+	create_zip
+	upload_zip_to_blob_storage
+	push_acs_branch
+	cleanup_output
+
+else
+	DIST_DIR=${zip_path}/k
+	create_dist_dir
+	build_kube_binaries_for_upstream_e2e
+	copy_dockerfile	
```

Comment:
I think this one is no longer needed after https://github.com/Azure/acs-engine/commit/87408da3fa70dfa43a9278e2c2d744c94d78fbfd, @PatrickLang?

Reply:
I tested yesterday with a zip without the Dockerfile and I couldn’t spawn pods on the windows nodes. Acs-engine v0.16.2